### PR TITLE
Alma branch

### PR DESCRIPTION
**Pull request by almaan**
### Submission type

* [ ] This submission adds a new dataset
* [x] This submission adds a new method
* [x] This submission adds a new metric
* [x] This submission adds a new task
* [ ] This submission adds a new Docker image
* [ ] This submission fixes a bug (link to related issue: )
* [ ] This submission adds a new feature not listed above

### Testing

* [x] This submission was written on a forked copy of SingleCellOpenProblems
* [x] GitHub Actions "Run Benchmark" tests are passing on this base branch of this pull request [link](https://github.com/giovp/SingleCellOpenProblems/runs/2225974506?check_suite_focus=true)
* [ ] If this pull request is not ready for review (including passing the "Run Benchmark" tests), I will open this PR as a draft (click on the down arrow next to the "Create Pull Request" button)

### Submission guidelines

* [x] This submission follows the guidelines in our [Contributing](../blob/master/CONTRIBUTING.md) document
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change

### Summary

To briefly summarize the updates.

* `README.md` written description of the datasets and task (could be updated later on)
* `api.py` : added `check_dataset, `check_method`,  `sample_dataset` and `sample_method`
* `metrics` : added the R2 metric using sklearn's [implementation](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html) 
* `methods` : added a method called `random.py` which randomly assigns proportions to each of the cell types, could be useful as a baseline.

### CODE REVIEW

File: openproblems/tasks/spatial_decomposition/api.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+def check_dataset(adata):
+    """Check that dataset output fits expected API."""
+    # test for spatial coordinates
+    assert "spatial" in adata.obsm
+    # check that proportions are included
+    assert "proportions_true" in adata.obsm
+    # make sure proportions sum to one, some precision error allowed
+    assert np.all(np.abs(np.sum(adata.obsm["proportions_true"], axis=1) - 1) < EPS)
```
Comment (hiraksarkar) [inline openproblems/tasks/spatial_decomposition/api.py:16]: In my experience converting to `float64` helps fix normalization error (not summing up to 1). We might consider doing that. For now, this works too.

File: openproblems/tasks/spatial_decomposition/__init__.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from ... import utils
+from . import api
+from . import datasets
+from . import methods
+from . import metrics
+
+_task_name = "Label Projection"
```
Comment (hiraksarkar) [inline openproblems/tasks/spatial_decomposition/__init__.py:7]: We want to change the task name.

**Review by hiraksarkar — CHANGES_REQUESTED** [review]
Looks good, we need to change the task name.

File: openproblems/tasks/spatial_decomposition/api.py (hunk 1 above)
Comment (almaan) [reply openproblems/tasks/spatial_decomposition/api.py:16]: I see your point, but would this also work if the provided dataset originally stored the values in float32 - my thought was that the loss of precision would already have occurred then and we wouldn't be able to retrieve more significant digits.

```python
np.random.seed(3)
arr1 = np.random.random(10).astype(np.float32)
arr2 = arr1 / arr1.sum()
arr3 = arr2.astype(np.float64)
print(arr3.sum())
```
gives me the value `0.9999999636784196` which implies that float-type conversion might not be sufficient in these cases.

File: openproblems/tasks/spatial_decomposition/__init__.py (hunk 2 above)
Comment (almaan) [reply openproblems/tasks/spatial_decomposition/__init__.py]: Well spotted, this is now taken care of!

File: openproblems/tasks/spatial_decomposition/api.py (hunk 1 above)
Comment (hiraksarkar) [reply openproblems/tasks/spatial_decomposition/api.py:16]: I see, yes, `np.all` is best in this case I guess.

**Review by hiraksarkar — APPROVED** [review]
